packages/cache/src/TargetHasher.ts

```diff
@@ -18,12 +29,28 @@ export class TargetHasher {
   constructor(private options: TargetHasherOptions) {}
 
   async hash(target: Target): Promise<string> {
+    const { root } = this.options;
+
     const hashKey = await salt(
       target.environmentGlob ?? this.options.environmentGlob ?? ["lage.config.js"],
       `${target.id}|${JSON.stringify(this.options.cliArgs)}`,
       this.options.root,
       this.options.cacheKey || ""
     );
+
+    if (target.cwd === root && target.cache) {
+      if (!target.inputs) {
+        throw new Error("Root-level targets must have `inputs` defined if it has cache enabled.");
+      }
+
+      const hashes = hashGlobGit(target.inputs, { cwd: root, gitignore: false }) ?? {};
+      const sortedHashMap = sortObject(hashes);
```

Comment:
is this sort needed? Seems like hashStrings sorts too.

Reply:
no it's not needed!